x-pack/plugins/security_solution/server/utils/promises_pool.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// TODO: taken from https://github.com/elastic/kibana/pull/121110 PR, ensure its not duplicated after merge
```

Comment:
That's very-very cool stuff, and I see why it is better than a simple chunk-and-promise-all approach, but could we please cover it with unit tests in either of the PRs? 🙏 
cc @xcrzx

Reply:
Covered in my PR